resource/chatadapter_test.go

```diff
@@ -0,0 +1,54 @@
+package resource_test
+
+import (
+	"context"
+	"os"
+	"testing"
+	"time"
+
+	"github.com/Pix4D/cogito/resource"
+)
+
+func TestChatAdapterMock(t *testing.T) {
+}
+
+func TestChatAdapterIntegration(t *testing.T) {
+	gchatUrl := os.Getenv("COGITO_TEST_GCHAT_HOOK")
+	if len(gchatUrl) == 0 {
+		t.Skip("Skipping integration test. See CONTRIBUTING for how to enable.")
+	}
+	job := "peel"
+
+	// We send multiple messages to better see the UI in GChat and run the subtests
+	// in parallel for speed.
+	testCases := []struct {
+		pipeline string
+		state    string
+	}{
+		{
+			pipeline: "avocado",
+			state:    "success",
+		},
+		{
+			pipeline: "coconut",
+			state:    "failure",
+		},
+	}
+
+	for _, tc := range testCases {
+		tc := tc // Capture range variable, needed for t.Parallel to work correctly.
+		t.Run(tc.pipeline, func(t *testing.T) {
+			t.Parallel()
```

Comment:
This ensures these tests cases are run in parallel?

Reply:
In this case, it requires all the subtests (there is a t.Run) to run in parallel. By default tests are run sequentially.